File: m2cgen/interpreters/go/softmax.go
func softmax(x []float64) []float64 {
    size := len(x)
    result := make([]float64, size)
    max := x[0]
    for _, v := range x {
        if (v > max) {
            max = v
        }
    }
    sum := 0.0
    for i := 0; i < size; i++ {
        result[i] = math.Exp(x[i] - max)
        sum += result[i]
    }
    for i := 0; i < size; i++ {
        result[i] /= sum
    }
    return result
}

File: m2cgen/interpreters/javascript/softmax.js
function softmax(x) {
    var size = x.length;
    var result = new Array(size);
    var max = x[0];
    for (var i = 1; i < size; ++i) {
        if (x[i] > max)
            max = x[i];
    }
    var sum = 0.0;
    for (var i = 0; i < size; ++i) {
        result[i] = Math.exp(x[i] - max);
        sum += result[i];
    }
    for (var i = 0; i < size; ++i)
        result[i] /= sum;
    return result;
}
